[CLX-1281][Student] - DRY The Title Bar

reabbotted opened the pull request:
refs: CLX-1281
builds: student
affects: student
release note: none

test plan: Affects the title bar on the following pages. Dashboard, Notebook, Notebook Note, Notifications, Inbox, Create Message, Skillspace, Account Profile, Account Notifications, Account Advanced.

I hope you're OK with my using AnyView here ....

reabbotted committed:
refs: CLX-1281
builds: student
affects: student
release note: none

test plan: Affects the title bar on the following pages. Dashboard, Notebook, Notebook Note, Notifications, Inbox, Create Message, Skillspace, Account Profile, Account Notifications, Account Advanced.

reabbotted committed:
including the rest

inst-danger commented:
<h3>Builds</h3><p>No apps were built for this pull request.</p><p><em>To trigger app builds, include a line starting with <code>builds:</code> followed by app names (Student, Teacher, Parent, or All) in your pull request's message.</em></p><p><strong>Commit:</strong> refs: CLX-1281 builds: student affects: student release note: none (<a href="https://github.com/instructure/canvas-ios/commit/f51e106">f51e106</a>)<br/><strong>Built At:</strong> Sep 05 19:07 CEST (09/05 11:07 AM MDT)</p>

<!-- bitrise_adhoc_builds -->

reabbotted committed:
More refactor work

reabbotted committed:
More refactor work

reabbotted committed:
Just adding some comments

szabinst reviewed (commented):
Can you check the paddings? I found inconsistencies between screens, check the center of the arrow: 
<img width="1956" height="958" alt="Screenshot 2025-09-09 at 13 46 18" src="https://github.com/user-attachments/assets/f16ea430-d069-4a41-a041-5f3b3ffa5207" />

reabbotted committed:
refs: CLX-1281
builds: student
affects: student
release note: none

test plan:

reabbotted committed:
Merge branch 'master' into CLX-1281-i-os-dry-the-navigation-bar-ui

reabbotted committed:
refs: CLX-1281
builds: student
affects: student
release note: none

test plan: Test the alignment of padding for the title bar on all pages.

reabbotted committed:
refs: CLX-1281
builds: student
affects: student
release note:

test plan: Test all of the pages with a title bar

szabinst committed:
fix: padding issues
